packages/java/gradle-plugin/src/main/kotlin/com/vaadin/hilla/gradle/plugin/EngineGenerateTask.kt

```diff
@@ -47,39 +44,36 @@ public open class EngineGenerateTask : DefaultTask() {
         }
     }
 
+    @Input
+    public val groupId: String = project.group.toString()
+
+    @Input
+    public val artifactId: String = project.name
+
+    @Input
+    @Optional
+    public var mainClass: String? = project.findProperty("mainClass") as String?
+
     @TaskAction
     public fun engineGenerate() {
         val extension: EngineProjectExtension = EngineProjectExtension.get(project)
         logger.info("Running the engineGenerate task with effective Hilla configuration $extension")
```

Comment:
Since the `exposedPackagesToParser` is no longer needed, I believe we can get rid of the `EngineProjectExtension` altogether, and move the utility method for creating the `EngineConfiguration` instance to some common parent of the tasks, or to some other utility class. Those extension objects were registered in the `HillaPlugin.kt` for extending the plugin's properties to support passing/configuring extra custom stuff from command-line or the build file.